File: app/src/main - 副本/java/com/zmcursor/mydaliy/UI/main/MainActivity.java
package com.zmcursor.mydaliy.UI.main;

import android.os.Bundle;
import android.os.Message;
import android.widget.Button;

import com.zmcursor.mydaliy.R;
import com.zmcursor.mydaliy.UI.base.BaseActivity;

public class MainActivity extends BaseActivity {

    public static final int msg_init = 354500;
    public static final int msg_refresh = 354511;
    public static final int msg_load_more = 354522;

    MainPresenter presenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        presenter = new MainPresenterImpl(this);


        Button more = findViewById(R.id.more);
        Button refresh = findViewById(R.id.refresh);

        more.setOnClickListener(v -> presenter.loadMore());

        refresh.setOnClickListener(v -> presenter.refresh());

    }

    @Override
    protected void onPause() {
        super.onPause();
        presenter.saveDate();
    }

    @Override
    public boolean handleMessage(Message msg) {
        switch (msg.what) {
            case msg_init:
                break;
            case msg_refresh:
                break;
            case msg_load_more:
                break;
            default:
                return false;
        }
        return true;
    }
}

File: app/src/main/java/com/zmcursor/mydaliy/UI/main/MainActivity.kt
package com.zmcursor.mydaliy.UI.main

import android.os.Bundle
import android.os.Message
import android.widget.Button

import com.zmcursor.mydaliy.R
import com.zmcursor.mydaliy.UI.base.BaseActivity

class MainActivity : BaseActivity() {

    internal var presenter: MainPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        presenter = MainPresenterImpl(this)


        val more = findViewById<Button>(R.id.more)
        val refresh = findViewById<Button>(R.id.refresh)

        more.setOnClickListener { v -> presenter.loadMore() }

        refresh.setOnClickListener { v -> presenter.refresh() }

    }

    override fun onPause() {
        super.onPause()
        presenter.saveDate()
    }

    override fun handleMessage(msg: Message): Boolean {
        when (msg.what) {
            msg_init -> {
            }
            msg_refresh -> {
            }
            msg_load_more -> {
            }
            else -> return false
        }
        return true
    }

    companion object {

        val msg_init = 354500
        val msg_refresh = 354511
        val msg_load_more = 354522
    }
}
